x/ccv/provider/keeper/relay.go

```diff
@@ -285,15 +292,19 @@ func (k Keeper) OnRecvSlashPacket(ctx sdk.Context, packet channeltypes.Packet, d
 func (k Keeper) ValidateSlashPacket(ctx sdk.Context, chainID string,
 	packet channeltypes.Packet, data ccv.SlashPacketData) error {
 
-	_, found := k.getMappedInfractionHeight(ctx, chainID, data.ValsetUpdateId)
 	// return error if we cannot find infraction height matching the validator update id
-	if !found {
-		return fmt.Errorf("cannot find infraction height matching "+
-			"the validator update id %d for chain %s", data.ValsetUpdateId, chainID)
+	if _, found := k.getMappedInfractionHeight(ctx, chainID, data.ValsetUpdateId); !found {
+		return providertypes.ErrSlashPacketInfractionHeightInvalid
 	}
 
```

Comment:
Drop SlashPackets for downtime infraction were received from the same consumer for infraction heights smaller than the latest downtime infraction height received from that consumer, e.g.,
```
if stakingtypes.Downtime
  if data.ValsetUpdateId < latest received data.ValsetUpdateId for same chain downtime
    return error
```
Not sure how important is this check and whether it is worth the trouble.

Reply:
Done.